cas-server-support-openid/src/main/java/org/jasig/cas/support/openid/web/support/OpenIdArgumentExtractor.java

```diff
@@ -1,39 +1,54 @@
- * Apereo licenses this file to you under the Apache License,
- * Version 2.0 (the "License"); you may not use this file
- * except in compliance with the License.  You may obtain a
- * copy of the License at the following location:
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing,
- * software distributed under the License is distributed on an
- * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
- * KIND, either express or implied.  See the License for the
- * specific language governing permissions and limitations
- * under the License.
- */
-package org.jasig.cas.support.openid.web.support;
-
-import org.jasig.cas.authentication.principal.WebApplicationService;
-import org.jasig.cas.support.openid.authentication.principal.OpenIdService;
-import org.jasig.cas.web.support.AbstractArgumentExtractor;
-
-import javax.servlet.http.HttpServletRequest;
-
-/**
- * Constructs an OpenId Service.
- *
- * @author Scott Battaglia
- * @since 3.1
- */
-public class OpenIdArgumentExtractor extends AbstractArgumentExtractor {
-
-    @Override
-    protected WebApplicationService extractServiceInternal(final HttpServletRequest request) {
-        return OpenIdService.createServiceFrom(request);
-    }
-}
+/*
+ * Licensed to Apereo under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work
+ * for additional information regarding copyright ownership.
+ * Apereo licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License.  You may obtain a
+ * copy of the License at the following location:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.jasig.cas.support.openid.web.support;
+
+import org.jasig.cas.authentication.principal.WebApplicationService;
+import org.jasig.cas.support.openid.authentication.principal.OpenIdService;
+import org.jasig.cas.web.support.AbstractArgumentExtractor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.NotNull;
+
+/**
+ * Constructs an OpenId Service.
+ *
+ * @author Scott Battaglia
+ * @since 3.1
+ */
+public class OpenIdArgumentExtractor extends AbstractArgumentExtractor {
+
+    /**
+     * The prefix url for OpenID (without the trailing slash).
+     */
+    @NotNull
+    private String openIdPrefixUrl;
+
+    @Override
+    protected WebApplicationService extractServiceInternal(final HttpServletRequest request) {
+        return OpenIdService.createServiceFrom(request, openIdPrefixUrl);
```

Comment:
An `openIdPrefixUrl` is added to be able to rebuild the full OpenID identifier: this prefix + the username taken from the authentication.